Scripts/Runtime/Logging/.CSProject/UnityLogListener.cs

```diff
@@ -179,10 +202,19 @@ public void LogFormat(LogType logType, UnityEngine.Object context, string format
                 return;
             }
 
-            SendToLogger(context, LogTypeToLogLevel(logType), string.Format(format, args));
+            // Make UnityEngine.Debug.Assert failures throw an exception instead of just logging an error
+            StackTrace stackTrace = new StackTrace(fNeedFileInfo: true);
+            MethodBase assertMethod = stackTrace.GetFrame(2)?.GetMethod(); // Skip this function and UnityEngine.Logger
+            if (assertMethod != null && assertMethod.Name == "Assert" && assertMethod.DeclaringType.FullName == "UnityEngine.Debug")
```

Comment:
A little bit safer
```suggestion
            if (assertMethod != null && assertMethod.Name == nameof(Debug.Assert) && assertMethod.DeclaringType.Assembly.ToString() == typeof(UnityEngine.Debug).Assembly.ToString())
```

Just for demonstrative purposes. You probably want to cache that `typeof`

Reply:
Now _this_ is my JS showing 😛 Can't believe I forgot to at least use `nameof(...)`